src/main/drivers/max7456.c

```diff
@@ -407,15 +430,28 @@ uint8_t* max7456GetScreenBuffer(void) {
 
 void max7456WriteChar(uint8_t x, uint8_t y, uint8_t c)
 {
-    screenBuffer[y*30+x] = c;
+    unsigned pos = y*30+x;
+    if (screenBuffer[pos] != c) {
+        screenBuffer[pos] = c;
+        bitArraySet(screenIsDirty, pos);
+    }
 }
 
 void max7456Write(uint8_t x, uint8_t y, const char *buff)
 {
     uint8_t i = 0;
-    for (i = 0; *(buff+i); i++)
-        if (x+i < 30) //do not write over screen
-            screenBuffer[y*30+x+i] = *(buff+i);
+    uint8_t c;
+    uint16_t pos;
+    for (i = 0; *buff; i++, buff++)
+        //do not write over screen
+        if (x+i < 30) {
+            c = *buff;
+            pos = y*30+x+i;
```

Comment:
`y*30+x` can be pre-calculated outside the loop

Reply:
Nice catch! We can also just increment `pos` once per iteration rather than always adding `i` and break out of the loop once `x+i >= 30`. Code should be cleaner this way too.